### ISSUE TITLE
Enhancement: lines-around-comment allowTupleStart

### ISSUE THREAD

**Issue by cassus** [typescript-eslint/typescript-eslint#6743]
### Before You File a Proposal Please Confirm You Have Done The Following...

- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22enhancement%3A+plugin+rule+option%22) and found none that match my proposal.
  - This would be a further extension of https://github.com/typescript-eslint/typescript-eslint/pull/5327
- [X] I have searched the [current rule list](https://typescript-eslint.io/rules/#supported-rules) and found no rules that match my proposal.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### My proposal is suitable for this project

- [X] I believe my proposal would be useful to the broader TypeScript community (meaning it is not a niche proposal).

### Link to the rule's documentation

https://typescript-eslint.io/rules/lines-around-comment/#options

### Description

We have allowInterfaceStart and allowTypeStart in lines-around-comment, would be nice if it also had allowTupleStart

### Fail

```typescript
/*eslint lines-around-comment: ["error", { "beforeBlockComment": true, "allowTupleStart": false }]*/

export type ATuple = [
  /* Comment that would be nice to be able to allow with eg. allowTupleStart (Gives error now) */
  1,
]
```

### Pass

```typescript
/*eslint lines-around-comment: ["error", { "beforeBlockComment": true, "allowTupleStart": true }]*/

export type ATuple = [
  /* Comment that would be nice to be able to allow with eg. allowTupleStart (Gives error now) */
  1,
]
```

### Playground link
https://typescript-eslint.io/play/#ts=5.0.2&sourceType=module&code=KYDwDg9gTgLgBDAnmYcCCAVArmANqgXjgG0AoOOAegCo4BhCAW0eADt4YALAQ3gHcIWXABM4AI1SsAlgGNUMCONTcx+BIu65cEPnD5SucYAHMAdHE3a+2PMADKMbrDjVK5OAEYANKQC6pANBIZyQUdAB5MQArYBl4IgBvdxo4ACFtGQBrOBkmFnY4TmAoVAEoTIBnOAAzKVZULAq64wstHQxke0dnV3cwKAgwAC5PUgBfUiA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6RplWgQ2gHtYTACa0yfALbjELdGADa4bDkTR+0SABpFSrEuyQARogBmfaIgAyzRAGEJUmRnxxEWvfu7x4fAO4AVEkQAZXxefFlnWERtbABfGIBdRXjYoA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA

**Comment by JoshuaKGoldberg** [comment]
Tuples are a really specific use case. And although I can see why it'd be valuable, we really don't have the bandwidth to maintain stylistic preferences as granular as this one. Per https://typescript-eslint.io/linting/troubleshooting/formatting#eslint-core-and-formatting we're also backing away from formatting/stylistic requests such as this one. 

My recommendation would be to either:

* _[suggested]_ Abandon hope of such specific stylistic requests, and embrace the zen of not linting for such things in code 😌 
* Fork the rule using our [custom rule docs](https://typescript-eslint.io/custom-rules)

Thanks for filing though!
